src/metadata.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone)]
+/// Metadata about a File or a Directory
+pub struct Metadata {
+    name: String,
+    size: u64,
+    created_time:  ::time::Tm,
+    modified_time: ::time::Tm,
+    user_metadata: Vec<u8>
+}
+
+impl Metadata {
+    /// Create a new instance of Metadata
+    pub fn new(name: String, user_metadata: Vec<u8>) -> Metadata {
+        Metadata {
+            name: name,
+            size: 0,
+            created_time:  ::time::now_utc(),
+            modified_time: ::time::now_utc(),
+            user_metadata: user_metadata
+        }
+    }
+
+    /// Get name associated with the structure (file or directory) that this metadata is a part
+    /// of
+    pub fn get_name(&self) -> &String {
+        &self.name
+    }
+
+    #[allow(dead_code)]
+    /// Set name associated with the structure (file or directory) that this metadata is a part
+    /// of
+    pub fn set_name(&mut self, name: String) {
+        self.name = name;
+    }
+
+    /// User setteble metadata for custom metadata
+    pub fn set_user_metadata(&mut self, user_metadata: Vec<u8>) {
+        self.user_metadata = user_metadata;
+    }
+
+    /// Get user setteble custom metadata
+    pub fn get_user_metadata(&self) -> Option<&Vec<u8>> {
+        if self.user_metadata.is_empty() {
```

Comment:
As of now user_metadata is vec<u8>. We can use combinators once we make it as Option. We have it as a todo probably by end of sprint, if time permits